wrapanapi/systems/lenovo.py

```diff
@@ -453,3 +456,143 @@ def get_device_unique_id(device):
+        'health_state:': lambda self, requester: self.get_switch_health_state(requester.name),
+        'type': lambda self, requester: self.get_switch_type(requester.name),
+        'ipv4_addresses': lambda self, requester: self.get_switch_ipv4_addresses(requester.name),
+        'ipv6_addresses': lambda self, requester: self.get_switch_ipv6_addresses(requester.name),
+        'manufacturer': lambda self, requester: self.get_switch_manufacturer(requester.name),
+        'description': lambda self, requester: self.get_switch_description(requester.name),
+        'serial_number': lambda self, requester: self.get_switch_serial_number(requester.name),
+        'part_number': lambda self, requester:
+            self.get_switch_part_number(requester.name),
+        'ipv4_assignments': lambda self, requester:
+            self.get_switch_ipv4_assignments(requester.name),
+        'ipv6_assignments': lambda self, requester:
+            self.get_switch_ipv6_assignments(requester.name),
+        'firmwares': lambda self, requester: self.get_switch_firmwares(requester.name),
+    }
+
+    def __init__(self, service_instance):
+        self._service_instance = service_instance
+
+    def switch_inventory(self, requester=None, *requested_items, **kwargs):
+
+        # Retrieve and return the switch inventory
+        requested_items = requested_items or self._switches_inventory_available
+
+        return {item: self._switches_inventory_available[item](self, requester)
+                for item in requested_items}
+
+    def _list_switches(self):
+        # Collect the switches associated with this LXCA
+        self._switches_list = self._service_instance["switchList"]
+        return self._service_instance["switchList"]
+
+    @property
+    def switches_list(self):
+        if not self._switches_list:
+            self._list_switches()
+        return self._switches_list
+
+    def get_switch(self, switch_name):
+        for switch in self._switches_list:
+            if 'name' in switch and switch['name'] == switch_name:
+                return switch
+        return None
+
+    def get_switch_product_name(self, switch_name):
+        switch = self.get_switch(switch_name)
+        return switch['productName']
+
+    def get_switch_uuid(self, switch_name):
+        switch = self.get_switch(switch_name)
+        return switch['uuid']
+
+    def get_switch_manufacturer(self, switch_name):
+        switch = self.get_switch(switch_name)
+        return switch['manufacturer']
+
+    def get_switch_serial_number(self, switch_name):
+        switch = self.get_switch(switch_name)
+        return switch['serialNumber']
+
+    def get_switch_part_number(self, switch_name):
+        switch = self.get_switch(switch_name)
+        return switch['partNumber']
+
+    def get_switch_description(self, switch_name):
+        switch = self.get_switch(switch_name)
+        return switch['description']
+
+    def get_switch_ipv4_assignments(self, switch_name):
```

Comment:
These methods doing more complex processing of the raw data do make sense to implement for convenience.